### ISSUE TITLE
Quick Deploy Fails to Start with Valid Job ID

### ISSUE THREAD

**Issue by mcarvin8** [forcedotcom/cli#3055]
### Summary
In our CI pipeline, we validate and then quick deploy Apex changes. We've had a script that has been working for over a year to parse the validate job log for the validate ID and then use that for the quick deployment. I've made no changes to this script when I upgraded our container to use the latest Salesforce CLI.

After testing the latest CLI build with the same script, I'm getting this error regarding the ID length
```
Error (InvalidIdLengthError): Parsing --job-id 
	The id must be 15 or 18 characters.
```

`0AfVG0000083Mdn0AE` is 18 characters. There isn't any hidden space between the command used : `sf project deploy quick -i 0AfVG0000083Mdn0AE -w 240`

### Steps To Reproduce

1. Run `sf project deploy validate` against Apex components
2. After successful validate, pass the validate ID to the `sf project deploy quick` command
3. Confirm quick deployment runs.

### Expected result
Quick deployment runs using the validate ID.

### Actual result
Quick deployment fails to start with a validate ID (18 characters).

### Additional information
![{72B1F85C-A7B7-4302-8EAA-7DB14244372B}](https://github.com/user-attachments/assets/d67db4ea-54b4-4cb9-af58-d493aea09519)

### System Information
Ubuntu Docker

```json
{
  "architecture": "linux-x64",
  "cliVersion": "@salesforce/cli/2.60.13",
  "nodeVersion": "node-v20.18.0",
  "osVersion": "Linux 6.1.87",
  "rootPath": "/usr/lib/node_modules/@salesforce/cli",
  "shell": "bash",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.2.4 (core)",
    "@oclif/plugin-commands 4.0.15 (core)",
    "@oclif/plugin-help 6.2.12 (core)",
    "@oclif/plugin-not-found 3.2.21 (core)",
    "@oclif/plugin-plugins 5.4.9 (core)",
    "@oclif/plugin-search 1.2.9 (core)",
    "@oclif/plugin-update 4.5.9 (core)",
    "@oclif/plugin-version 2.2.13 (core)",
    "@oclif/plugin-warn-if-update-available 3.1.17 (core)",
    "@oclif/plugin-which 3.2.14 (core)",
    "@salesforce/cli 2.60.13 (core)",
    "apex 3.4.10 (core)",
    "api 1.2.2 (core)",
    "auth 3.6.62 (core)",
    "data 3.6.7 (core)",
    "deploy-retrieve 3.12.11 (core)",
    "info 3.4.9 (core)",
    "limits 3.3.32 (core)",
    "marketplace 1.2.26 (core)",
    "org 4.5.10 (core)",
    "packaging 2.8.7 (core)",
    "schema 3.3.32 (core)",
    "settings 2.3.21 (core)",
    "sobject 1.4.38 (core)",
    "source 3.5.21 (core)",
    "telemetry 3.6.15 (core)",
    "templates 56.3.21 (core)",
    "trust 3.7.31 (core)",
    "user 3.5.31 (core)",
    "sf-decomposer 5.0.0 (user) published 3 days ago (Fri Oct 04 2024)",
    "sfdx-git-delta 5.47.0 (user) published 0 days ago (Tue Oct 08 2024)"
  ]
}
```

**Comment by mcarvin8** [comment]
Issue still present with latest CLI.

![{422638CC-C0AA-4122-926C-42B9B2B5F4B5}](https://github.com/user-attachments/assets/067f52d6-86aa-4cb6-88e2-9674a370b692)


``` json
{
  "architecture": "linux-x64",
  "cliVersion": "@salesforce/cli/2.61.8",
  "nodeVersion": "node-v20.18.0",
  "osVersion": "Linux 6.1.87",
  "rootPath": "/usr/lib/node_modules/@salesforce/cli",
  "shell": "bash",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.2.5 (core)",
    "@oclif/plugin-commands 4.0.16 (core)",
    "@oclif/plugin-help 6.2.13 (core)",
    "@oclif/plugin-not-found 3.2.22 (core)",
    "@oclif/plugin-plugins 5.4.10 (core)",
    "@oclif/plugin-search 1.2.10 (core)",
    "@oclif/plugin-update 4.5.10 (core)",
    "@oclif/plugin-version 2.2.14 (core)",
    "@oclif/plugin-warn-if-update-available 3.1.18 (core)",
    "@oclif/plugin-which 3.2.15 (core)",
    "@salesforce/cli 2.61.8 (core)",
    "apex 3.5.0 (core)",
    "api 1.2.2 (core)",
    "auth 3.6.65 (core)",
    "data 3.6.8 (core)",
    "deploy-retrieve 3.12.15 (core)",
    "info 3.4.9 (core)",
    "limits 3.3.32 (core)",
    "marketplace 1.2.26 (core)",
    "org 4.6.0 (core)",
    "packaging 2.8.10 (core)",
    "schema 3.3.34 (core)",
    "settings 2.3.23 (core)",
    "sobject 1.4.40 (core)",
    "source 3.5.21 (core)",
    "telemetry 3.6.15 (core)",
    "templates 56.3.21 (core)",
    "trust 3.7.32 (core)",
    "user 3.5.32 (core)",
    "sf-decomposer 5.0.0 (user) published 6 days ago (Fri Oct 04 2024) (latest is 5.0.1)",
    "sfdx-git-delta 5.47.0 (user) published 3 days ago (Tue Oct 08 2024)"
  ]
}
```

**Comment by mdonnalley** [comment]
> We've had a script that has been working for over a year to parse the validate job log for the validate ID and then use that for the quick deployment.

I'm not able to replicate locally without knowing what your script is doing to find the job id

Also, we **strongly** discourage parsing human readable output since that output could change at any point without notice. For that reason, we recommend using `--json` if you need to access things like job ids. 

Lastly, as a workaround you can use the `--use-most-recent` flag to do a quick deploy of the most recently validated deployment. That would remove the need to parse the job id

**Comment by mcarvin8** [comment]
I'd rather not use the `--json` format since that removes the deploy progress bars and our developers would probably not react well if we removed them. If there's a way to use `--json` format while retaining the progress bar output, that would be nice.

I'm fine with using the `--use-most-recent` flag to work around this for now, but I preferred the validate ID used directly to avoid any potential issues with using the wrong ID.

Our pipeline runs the validate and quick-deploy in the same pipeline and we use Resource Groups to avoid parallel deployments, but I'm hoping there isn't some weird case where a different validate ID was the most recent deployment.

I'm fine with just closing this for now, I'm testing an update to the script now to see if there was an hidden spaces in the newer output. All the script is doing is re-directing the `sf project deploy start` output to a text file and the script reads that text file for `Deploy Id: *`. The `sf project deploy start` output, with the deploy bars, are still printed in real time, but I'm using `tee` to send the output to a text file in parallel.

**Comment by Faizahamed-dev** [comment]
Error (InvalidIdLengthError): Parsing --job-id 
	The id must be 15 or 18 characters.
See more help with --help  my deploy id is having 18 digit 
i am using the command please let me know if anything missing here sf project deploy report -i "$deploy_id" -u "${SALESFORCE_USERNAME}" --wait 0 --use-most-recent as well  sf project deploy report -i "$deploy_id" -u "${SALESFORCE_USERNAME}" --wait 0 --json

**Comment by mcarvin8** [comment]
@Faizahamed-dev - If you're still having this issue, I would recommend creating a new issue in this repo with your details.

I ended up switching to just `sf project deploy quick --use-most-recent -w $DEPLOY_TIMEOUT` to always the use latest validate without needing the validate id.

But using the validate Id should be possible and was working fine for me for the longest time until I upgraded CLI builds.
